### ISSUE TITLE
Documentation of file format guessing is missing

### ISSUE THREAD

**Issue by volter** [logstash-plugins/logstash-filter-translate#25]
Actually the JSON and CSV capabilities are not documented at all. Expecting a certain file extension broke my setup, when I updated this plug-in.

I would rather see the file format as a configuration setting than as a guess.

**Comment by pemontto** [comment]
:+1:

**Comment by magnusbaeck** [comment]
This autodetection is documented in the [source code](https://github.com/logstash-plugins/logstash-filter-translate/blob/v2.1.3/lib/logstash/filters/translate.rb#L59-L74),

> Currently supported formats are YAML, JSON and CSV, format selection is based on the file extension, json for JSON, (yaml|yml) for YAML and csv for CSV.

but this hasn't reached the rendered documentation at elastic.co. Perhaps a documentation generation bug? Alternatively, the [documentation rendering of the master branch](https://www.elastic.co/guide/en/logstash/master/plugins-filters-translate.html) won't automatically pick up changes in the master branch of this plugin.

**Comment by geekpete** [comment]
Looks like something odd going on, either way the documentation isn't reflective of the plugin's current capabilities.

I have tested that all three dictionary formats work for me - csv, json, yaml:

Example data to show the formats:
```
==> dictionary.csv <==
"Person1","Henry"
"Person2","Thomas"
"Person3","Rufus"

==> dictionary.json <==
{
"Person1": "Henry"
"Person2": "Thomas"
"Person3": "Rufus"
}

==> publishers.yml <==
"Person1": "Henry"
"Person2": "Thomas"
"Person3": "Rufus"
```

**Comment by PhaedrusTheGreek** [comment]
The source code for the documentation actually looks like this, and contains the explanation of how to handle file extensions, but anything after the first NOTE is not displaying in the [Logstash Documentation](https://www.elastic.co/guide/en/logstash/master/plugins-filters-translate.html#plugins-filters-translate-dictionary_path)

```
  # NOTE: it is an error to specify both `dictionary` and `dictionary_path`
  # NOTE: Currently supported formats are YAML, JSON and CSV, format selection is
  # based on the file extension, json for JSON, (yaml|yml) for YAML and csv for CSV.
  # NOTE: The JSON format only supports simple key/value, unnested objects. The CSV
  # format expects exactly two columns with the first serving as the original text,
  # the second column as the replacement
```

![screen shot 2016-06-16 at 1 01 32 pm](https://cloud.githubusercontent.com/assets/4387023/16125635/88a6d446-33c2-11e6-8f11-63a6b41a7721.png)

Related Issue:
https://github.com/elastic/logstash/issues/5261

@suyograo This might be something that needs to be fixed in the Logstash Documentation conversion?

**Comment by dedemorton** [comment]
Fixed in #40
